br/pkg/task/operator/cmd.go

```diff
@@ -0,0 +1,101 @@
+package operator
+
+import (
+	"context"
+	"crypto/tls"
+	"strings"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/br/pkg/logutil"
+	"github.com/pingcap/tidb/br/pkg/pdutil"
+	"github.com/pingcap/tidb/br/pkg/task"
+	"github.com/pingcap/tidb/br/pkg/utils"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+func dialPD(ctx context.Context, cfg *task.Config) (*pdutil.PdController, error) {
+	pdAddrs := strings.Join(cfg.PD, ",")
+	var tc *tls.Config
+	if cfg.TLS.IsEnabled() {
+		var err error
+		tc, err = cfg.TLS.ToTLSConfig()
+		if err != nil {
+			return nil, err
+		}
+	}
+	mgr, err := pdutil.NewPdController(ctx, pdAddrs, tc, cfg.TLS.ToPDSecurityOption())
+	if err != nil {
+		return nil, err
+	}
+	return mgr, nil
+}
+
+func cleanUpWith(f func(ctx context.Context)) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	f(ctx)
+}
+
+// PauseGCAndScheduler blocks the current goroutine and pause the GC safepoint and remove the scheduler by the config.
+// This function will block until the context being canceled.
+func PauseGCAndScheduler(ctx context.Context, cfg *PauseGcConfig) error {
+	mgr, err := dialPD(ctx, &cfg.Config)
+	if err != nil {
+		return err
+	}
+
+	eg, ectx := errgroup.WithContext(ctx)
+
+	eg.Go(func() error { return pauseGCKeeper(ectx, cfg, mgr) })
+	eg.Go(func() error { return pauseSchedulerKeeper(ectx, mgr) })
+
+	return eg.Wait()
+}
+
+func pauseGCKeeper(ctx context.Context, cfg *PauseGcConfig, ctl *pdutil.PdController) error {
+	// Note: should we remove the service safepoint as soon as this exits?
+	sp := utils.BRServiceSafePoint{
+		ID:       utils.MakeSafePointID(),
+		TTL:      int64(cfg.TTL.Seconds()),
```

Comment:
`GC` has a ttl, but `scheduler` resume as soon as function exits. is this work expected?

Reply:
I think blocking GC for a little longer won't bring bad things. (But it will probably be a disaster when workloads are applied to a cluster without schedulers...)